roles/sap_swpm/tasks/pre_install/assert_hostname_resolution_for_ha.yml

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+---
+
+# Assert HA settings
+
+- name: SAP SWPM Pre Install - HA settings - Ensure the dig command is present
+  ansible.builtin.package:
+    name: bind-utils
+    state: present
+
+- name: SAP SWPM Pre Install - HA settings - Try to resolve sap_swpm_virtual_hostname from DNS
+  ansible.builtin.command: dig +short +tries=1 +time=1 "{{ sap_swpm_virtual_hostname }}"
+  register: __sap_swpm_register_virtual_ip_dns
+  changed_when: false
+  failed_when: false
+
+- name: SAP SWPM Pre Install - HA settings - Assign sap_swpm_virtual_ip from DNS
+  ansible.builtin.set_fact:
+    __sap_swpm_fact_virtual_ip: "{{ __sap_swpm_register_virtual_ip_dns.stdout_lines[-1] }}"
+  when: __sap_swpm_register_virtual_ip_dns.stdout_lines | length > 0
+
+- name: SAP SWPM Pre Install - HA settings - Try using /etc/hosts for name resolution
+  when: __sap_swpm_register_virtual_ip_dns.stdout_lines | length == 0
+  block:
+
+    - name: SAP SWPM Pre Install - HA settings - Try to resolve sap_swpm_virtual_hostname from /etc/hosts
+      ansible.builtin.shell: |
+        awk '/\s{{ sap_swpm_virtual_hostname }}\./||/\s{{ sap_swpm_virtual_hostname }}/{print $1}' /etc/hosts
```

Comment:
- We don't need `set -o pipefail` because there is no pipe in the shell command.
- Searching for a string without the surrounding white spaces or a following dot (`.`) is not sufficient because we want to search for the exact hostname. I agree on excluding the comment character in column 1 so we do not catch lines which are part of a comment.
- We can accept an empty string as the result of the `awk` command because the task `Assert that sap_swpm_virtual_hostname can be resolved` will fail with the message `FAIL: HOST cannot be resolved` if `__sap_swpm_fact_virtual_ip` is empty after the previous task (`HOST` showing the hostname to be resolved).